### PR TITLE
JAMES-2070 LDAP should work after restart

### PR DESCRIPTION
**Pull request by chibenwa**
Includes integration test and simple fix!

I also had a problem running LDAP integration tests, as discussed with @aduprat , and I updated the testcontainers library, which solved the issue.

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[Building Java 8 james-2070-a767af8f-c6c2-4d0e-9712-0f0d15287cee] Tests in error: 
[Building Java 8 james-2070-a767af8f-c6c2-4d0e-9712-0f0d15287cee]   TikaTextExtractorTest.setUp:58 » NoSuchMethod org.testcontainers.containers.Ge...
[Building Java 8 james-2070-a767af8f-c6c2-4d0e-9712-0f0d15287cee]   TikaTextExtractorTest.setUp:58 » NoSuchMethod org.testcontainers.containers.Ge...
[Building Java 8 james-2070-a767af8f-c6c2-4d0e-9712-0f0d15287cee]   TikaTextExtractorTest.setUp:58 » NoSuchMethod org.testcontainers.containers.Ge...
[Building Java 8 james-2070-a767af8f-c6c2-4d0e-9712-0f0d15287cee]   TikaTextExtractorTest.setUp:58 » NoSuchMethod org.testcontainers.containers.Ge...
[Building Java 8 james-2070-a767af8f-c6c2-4d0e-9712-0f0d15287cee]   TikaTextExtractorTest.setUp:58 » NoSuchMethod org.testcontainers.containers.Ge...
[Building Java 8 james-2070-a767af8f-c6c2-4d0e-9712-0f0d15287cee]   TikaTextExtractorTest.setUp:58 » NoSuchMethod org.testcontainers.containers.Ge...
[Building Java 8 james-2070-a767af8f-c6c2-4d0e-9712-0f0d15287cee]   TikaTextExtractorTest.setUp:58 » NoSuchMethod org.testcontainers.containers.Ge...
[Building Java 8 james-2070-a767af8f-c6c2-4d0e-9712-0f0d15287cee]   TikaTextExtractorTest.setUp:58 » NoSuchMethod org.testcontainers.containers.Ge...
[Building Java 8 james-2070-a767af8f-c6c2-4d0e-9712-0f0d15287cee]   TikaTextExtractorTest.setUp:58 » NoSuchMethod org.testcontainers.containers.Ge...
[Building Java 8 james-2070-a767af8f-c6c2-4d0e-9712-0f0d15287cee]   TikaTextExtractorTest.setUp:58 » NoSuchMethod org.testcontainers.containers.Ge...
[Building Java 8 james-2070-a767af8f-c6c2-4d0e-9712-0f0d15287cee]   TikaTextExtractorTest.setUp:58 » NoSuchMethod org.testcontainers.containers.Ge...
[Building Java 8 james-2070-a767af8f-c6c2-4d0e-9712-0f0d15287cee]   TikaTextExtractorTest.setUp:58 » NoSuchMethod org.testcontainers.containers.Ge...
[Building Java 8 james-2070-a767af8f-c6c2-4d0e-9712-0f0d15287cee]   TikaTextExtractorTest.setUp:58 » NoSuchMethod org.testcontainers.containers.Ge...
[Building Java 8 james-2070-a767af8f-c6c2-4d0e-9712-0f0d15287cee]   TikaTextExtractorTest.setUp:58 » NoSuchMethod org.testcontainers.containers.Ge...
[Building Java 8 james-2070-a767af8f-c6c2-4d0e-9712-0f0d15287cee]   TikaTextExtractorTest.setUp:58 » NoSuchMethod org.testcontainers.containers.Ge...
```

Requires fixing. I will do this this afternoon.

File: server/data/data-ldap-integration-testing/src/test/java/org/apache/james/user/ldap/LdapGenericContainer.java [hunk 1]
```diff
@@ -85,8 +85,8 @@ public void stop() {
 
     public String getLdapHost() {
         return "ldap://" +
-                container.getContainerIpAddress() +
+                container.getIp() +
```
Comment (strokyl) [inline server/data/data-ldap-integration-testing/src/test/java/org/apache/james/user/ldap/LdapGenericContainer.java]: So this now work! At least

Comment (chibenwa) [inline server/data/data-ldap-integration-testing/src/test/java/org/apache/james/user/ldap/LdapGenericContainer.java]: Wait, I have problems on CI :trollface:

File: server/data/data-ldap-integration-testing/src/test/java/org/apache/james/user/ldap/ReadOnlyUsersLDAPRepositoryTest.java [hunk 2]
```diff
@@ -121,13 +121,31 @@ public void unknownUserShouldNotBeAbleToLogInWhenPasswordIsCorrect() throws Exce
...
+        startUsersRepository(ldapRepositoryConfigurationWithVirtualHosting());
+        ldapRepository.test(JAMES_USER_MAIL, PASSWORD);
+
+        ldapContainer.stop();
+        try {
+            ldapRepository.test(JAMES_USER_MAIL, PASSWORD);
+        } catch (Exception e) {
+            LOGGER.info("This exception is expected as we shut down the LDAP and forced its use", e);
+        }
+        setup();
```
Comment (mbaechler) [inline server/data/data-ldap-integration-testing/src/test/java/org/apache/james/user/ldap/ReadOnlyUsersLDAPRepositoryTest.java]: you should not call setup, you should have a method to restart the ldap container, but calling setup is unpredictable in the future

Comment (chibenwa) [inline server/data/data-ldap-integration-testing/src/test/java/org/apache/james/user/ldap/ReadOnlyUsersLDAPRepositoryTest.java]: Ok

File: backends-common/elasticsearch/pom.xml [hunk 3]
```diff
@@ -178,7 +178,7 @@
                 <dependency>
                     <groupId>org.testcontainers</groupId>
                     <artifactId>testcontainers</artifactId>
-                    <version>1.1.7</version>
+                    <version>1.3.1</version>
```
Comment (aduprat) [inline backends-common/elasticsearch/pom.xml]: scope is missing

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
```
[Building Java 8 james-2070-fd7d7be3-cbd5-4204-aae4-d119e5e1e802] Tests in error: 
[Building Java 8 james-2070-fd7d7be3-cbd5-4204-aae4-d119e5e1e802]   CassandraNodeConfTest.serverShouldStartWhenOneCassandraNodeIsUnreachable:82->assertThatServerStartCorrectly:95 » Provision
```

test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**